packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -3010,6 +3017,32 @@ export class SnapController extends BaseController<
     );
   }
 
+  /**
+   * Validate that the platform version specified in the manifest (if any) is
+   * compatible with the current platform version.
+   *
+   * @param manifest - The Snap manifest.
+   * @throws If the platform version is greater than the current platform
+   * version.
+   */
+  #validatePlatformVersion(manifest: SnapManifest) {
+    if (manifest.platformVersion === undefined) {
+      return;
+    }
+
+    if (semver.gt(manifest.platformVersion, getPlatformVersion())) {
```

Comment:
Patch versions are probably fine and may not need to reject, WDYT? 🤔

Reply:
Technically we can't guarantee that it'll work on a newer patch version, I think, so to be safe, I decided to reject any newer versions.